apps/src/lib/node/ledger/shell/prepare_proposal/block_space_alloc.rs

```diff
@@ -0,0 +1,254 @@
+//! Primitives that facilitate keeping track of the number
+//! of bytes utilized by some Tendermint consensus round's proposal.
+//!
+//! This is important, because Tendermint places an upper bound
+//! on the size of a block, rejecting blocks whose size exceeds
+//! the limit stated in [`RequestPrepareProposal`].
+//!
+//! The code in this module doesn't perform any deserializing to
+//! verify if we are, in fact, allocating space for the correct
+//! kind of tx for the current [`BlockSpaceAllocator`] state. It
+//! is up to the user to dispatch the correct kind of tx into the
+//! current state of the allocator.
+//!
+//! # How space is allocated
+//!
+//! In the current implementation, we allocate space for transactions
+//! in the following order of preference:
+//!
+//! - First, we allocate space for DKG decrypted txs.
+//! - Next, we allocate space for protocol txs. Protocol txs get 1/3 of the
+//!   block space allotted to them.
```

Comment:
```suggestion
//! - Next, we allocate space for protocol txs. 
```
"Protocol txs get 1/3 of the block space allotted to them" - this is a bit ambiguous to me? Should it be that protocol txs get allocated 1/3 of the total block space?

Reply:
```
allotted  -> tendermint block space chunk
allocated -> bin space chunk
```
we allot tendermint block space chunks to tx bins, and allocate tx bin chunks for txs